opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -66,7 +67,8 @@ namespace Opm {
             AllProperties = IntProperties | DoubleProperties
         };
 
-        EclipseState(std::shared_ptr< const Deck > deck , const ParseContext& parseContext);
+        EclipseState(std::shared_ptr< const Deck > );
```

Comment:
EclipseGrid and EclipseState now have constructors accepting references to Deck instead of shared_ptr's.

It took some refactoring;  I had to tweak the constructors of NNC, Schedule, and InitConfit to accept Deck& instead of DeckPtr as well.

But all in all, I think this is a good change that we should cherry-pick if we don't want this entire PR.